dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/user/UserController.java

```diff
@@ -817,14 +821,33 @@ protected void prePatchEntity( User oldEntity, User newEntity )
     }
 
     @Override
-    protected void postPatchEntity( User user )
+    protected void postJsonPatch( JsonPatch patch, User user )
     {
         // Make sure we always expire all the user's active sessions if we
         // have disabled the user.
         if ( user != null && user.isDisabled() )
         {
             userService.expireActiveSessions( user );
         }
+
+        // Support patching user.userGroups relation which User is not the owner
+        for ( JsonPatchOperation op : patch.getOperations() )
+        {
+            JsonPointer userGroups = op.getPath().matchProperty( "userGroups" );
+            if ( userGroups == null )
+            {
+                continue;
+            }
+
+            String opName = op.getOp();
+            if ( opName.equals( "add" ) )
```

Comment:
Could we use some sort of constant here?

Reply:
Fixed